ArduCopter/radio.cpp

```diff
@@ -139,7 +139,7 @@ void Copter::set_throttle_and_failsafe(uint16_t throttle_pwm)
     if (throttle_pwm < (uint16_t)g.failsafe_throttle_value) {
 
         // if we are already in failsafe or motors not armed pass through throttle and exit
-        if (failsafe.radio || !(ap.rc_receiver_present || motors.armed())) {
+        if (failsafe.radio || !(ap.rc_receiver_present || hal.util->get_soft_arm_state() != AP_HAL::Util::SOFT_ARM_STATE_DISARMED)) {
```

Comment:
Actually you'd have to invert the contents of the parenthesis using De Morgan's law, which means the || would change to an && and the parenthesis would need to stay:
```
(!ap.rc_receiver_present && hal.util->get_soft_arm_state() == AP_HAL::Util::SOFT_ARM_STATE_DISARMED)
```
So: if(radio failsafe || disarmed without a radio present). Does that logic even look right to you considering the comment is "if we are already in failsafe or motors not armed pass through throttle and exit?" @rmackay9 can we update the comment to clarify the intention?